lab11_2/src/main.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { useState } from 'react';
+
+import './style.css';
+
+const App = () => {
+  const [time, setTime] = useState(0);
+  return (
+    <div id="app-container">
+      <label>Hours:</label>
+      <input value={unConvert(time, 'h')} onChange={e => setTime(converter(e.target.value, 'h'))}/>
+      <br></br>
+      <label>Minutes:</label>
+      <input value={unConvert(time, 'm')} onChange={e => setTime(converter(e.target.value, 'm'))}/>
+      <br></br>
+      <label>Seconds:</label>
+      <input value={unConvert(time, 's')} onChange={e => setTime(converter(e.target.value, 's'))}/>
+      <br></br>
+    </div>
+  );
+
+  function unConvert(val, identifier) {
+    return {
+      h: val / 60 / 60,
+      m: val / 60,
+      s: val
+    }[identifier];

```

Comment:
Just a small aside on this switch/case - note that you calculate ALL branches even though you only need one. It's not a problem with constants, it's a bit more work with math and it would be terrible if you actually called external functions that do something in the branches you didn't mean to visit